### PR TITLE
update Go (1.23), BuildKit (0.15.2), RootlessKit (2.3.1), and test deps

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Will release v2.0.0-rc.1 after merging this

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
Lint failures
```
Error: pkg/composer/run.go:132:5: S1009: should omit nil check; len() for []string is defined as zero (gosimple)
  	if ro.Volume != nil && len(ro.Volume) > 0 {
  	   ^
  Error: pkg/composer/run.go:141:5: S1009: should omit nil check; len() for []string is defined as zero (gosimple)
  	if ro.Entrypoint != nil && len(ro.Entrypoint) > 0 {
  	   ^
  Error: pkg/composer/run.go:145:5: S1009: should omit nil check; len() for []string is defined as zero (gosimple)
  	if ro.Env != nil && len(ro.Env) > 0 {
  	   ^
  Error: pkg/testutil/testutil.go:499:22: printf: non-constant format string in call to fmt.Errorf (govet)
  			return fmt.Errorf(msg)
  			                  ^
  Error: pkg/logging/cri_logger_test.go:93:14: printf: non-constant format string in call to (*testing.common).Fatalf (govet)
  				t.Fatalf(err.Error())
  				         ^
  Error: pkg/logging/json_logger_test.go:163:14: printf: non-constant format string in call to (*testing.common).Fatalf (govet)
  				t.Fatalf(err.Error())
  				         ^
  Error: cmd/nerdctl/compose_exec_linux_test.go:101:47: SA40[32](https://github.com/containerd/nerdctl/actions/runs/10433816274/job/28895988699?pr=3320#step:4:34): due to the file's build constraints, runtime.GOOS will never equal "windows" (staticcheck)
  		if !strings.Contains(stdout, "\nBAZ=\n") && runtime.GOOS != "windows" {
  		                                            ^
  Error: cmd/nerdctl/compose_exec_linux_test.go:116:50: SA4032: due to the file's build constraints, runtime.GOOS will never equal "windows" (staticcheck)
  		if !strings.Contains(stdout, "\nGARPLY=\n") && runtime.GOOS != "windows" {
  		                                               ^
  Error: cmd/nerdctl/compose_exec_linux_test.go:119:49: SA[40](https://github.com/containerd/nerdctl/actions/runs/10433816274/job/28895988699?pr=3320#step:4:42)32: due to the file's build constraints, runtime.GOOS will never equal "windows" (staticcheck)
  		if !strings.Contains(stdout, "\nWALDO=\n") && runtime.GOOS != "windows" {
```

**Comment by apostasie** [comment]
@AkihiroSuda we should add the lint tests to the canary target as well (right now, it only does tests) so that we can catch these things earlier.

Will do that, along with lint fixes.

File: Dockerfile [hunk 1]
```diff
@@ -44,17 +44,17 @@ ARG TINI_VERSION=v0.19.0
...
 ARG GOTESTSUM_VERSION=v1.12.0
 ARG NYDUS_VERSION=v2.2.5
-ARG SOCI_SNAPSHOTTER_VERSION=0.6.1
+ARG SOCI_SNAPSHOTTER_VERSION=0.7.0
 
 FROM --platform=$BUILDPLATFORM tonistiigi/xx:1.4.0 AS xx
 
 
-FROM --platform=$BUILDPLATFORM golang:${GO_VERSION}-bullseye AS build-base-debian
+FROM --platform=$BUILDPLATFORM golang:${GO_VERSION}-bookworm AS build-base-debian
```
Comment (AkihiroSuda) [inline Dockerfile:57]:
```
 > [linux/amd64->arm64 build-runc 4/4] RUN GO=xx-go make static &&   xx-verify --static runc && cp -v -a runc /out/runc.arm64:
0.397 xx-go build -trimpath -buildmode=pie  -tags "seccomp netgo osusergo" -ldflags "-X main.gitCommit=v1.1.13-0-g58aa9203 -X main.version=1.1.13 -linkmode external -extldflags --static-pie " -o runc .
99.10 # github.com/seccomp/libseccomp-golang
99.10 # [pkg-config --cflags  -- libseccomp]
99.10 Package libseccomp was not found in the pkg-config search path.
99.10 Perhaps you should add the directory containing `libseccomp.pc'
99.10 to the PKG_CONFIG_PATH environment variable
99.10 Package 'libseccomp', required by 'virtual:world', not found
137.5 make: *** [Makefile:69: static] Error 1
------
WARNING: No output specified with docker-container driver. Build result will only remain in the build cache. To push result image into registry use --push or to load image into docker use --load
Dockerfile:93
--------------------
  92 |     ENV CGO_ENABLED=1
  93 | >>> RUN GO=xx-go make static && \
  94 | >>>   xx-verify --static runc && cp -v -a runc /out/runc.${TARGETARCH}
  95 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c GO=xx-go make static &&   xx-verify --static runc && cp -v -a runc /out/runc.${TARGETARCH}" did not complete successfully: exit code: 2
```

https://github.com/containerd/nerdctl/actions/runs/10434592893/job/28897634369?pr=3320

**Review by djdongjin — APPROVED** [review]
LGTM, thanks

**Pull request merged**